services/spar/src/Spar/SCIM.hs

```diff
@@ -271,11 +272,17 @@ instance SCIM.UserDB Spar where
 
     buid <- Id <$> liftIO UUID.nextRandom
     -- TODO: Assume that externalID is the subjectID, let's figure out how
-    -- to extract that later. The question of "who's the issuer?" is a very
-    -- tricky one and we don't have a decision here yet.
-    let uref = SAML.UserRef
-          (SAML.Issuer $ SAML.unsafeParseURI ("https://unknown.issuer.org"))
-          (SAML.opaqueNameID extId)
+    -- to extract that later
+    -- TODO: when the issuer is deleted, the token still remains, so we can
+    -- fail here
+    -- TODO: throw SCIM errors here
+    issuer <- case mbIdp of
+        Nothing -> error "No IdP configured for the provisioning token"
+        -- TODO: should 'wrapMonadClient' or 'wrapMonadClientWithEnv' be used here?
```

Comment:
`wrapMonadClientWithEnv` gives you more context than `MonadClient`, so only use it if you need that context.  Currently `Env` carries the current system time and two SAML-specific TTL-values.